WordPress/gutenberg
# Experiment: Add lightbox to Image block using directives

## What?
This is part of the experiments section with the interactivity API in revising core blocks.
It uses directives from the [Directives hydration system](https://github.com/WordPress/block-interactivity-experiments/issues/80)

## Why?
It would enable native support for image lightboxes, a commonly used feature in many websites, without the use of jQuery or other external libraries.

## How?
It adds directives on click to images, presenting a larger version of the image along with an overlay, which can be hidden with an additional click, by scrolling, or pressing the escape key.

## Testing Instructions
1. Add an image to a post
2. Test enabling and disabling the lightbox using image's Advanced panel
3. View the post
4. Click on the image

### Testing Instructions for Keyboard

1. Tab to the image.
2. Press Enter to open the lightbox.
3. Test Tab or Shift + Tab — the lightbox should close and focus should go back to the image.
4. Also test pressing Escape to close the lightbox.

## Notes
- This is an early implementation; the idea is for themes to be able to enable or disable the lightbox behavior, for child themes to be able to override that behavior, and for users to be able to toggle lightboxes for all images sitewide or at the block level.
- This is part of a broader initiative to enable adding Interactivity API behaviors to components using the editor.
- Currently, we can just toggle the lightbox for a particular image at the block level under the Advanced panel (see below)
- The CSS implementation is not finalized. Ideally, I'd like to calculate the screen size and set the image to say 80% of the screen width, which will likely require some calculation outside of CSS. Is there a best practice for doing this?
- I'm looking for feedback on the design and overall approach

## Screenshots or screencast 

### Editor

https://user-images.githubusercontent.com/5360536/232110768-4749230c-facd-4d3e-b7b4-c46eaddf8b6d.mp4

### Frontend

https://user-images.githubusercontent.com/5360536/230152865-5957d4b0-dd71-41ad-b54d-0f1e4490c991.mp4
